src/Layout.js

```diff
@@ -1122,8 +1122,53 @@ var computeLayout = (function() {
       child.nextAbsoluteChild = null;
     }
   }
+  
+  function layoutNode(node, parentMaxWidth, parentDirection) {
+    node.shouldUpdate = true;
+    
+    var direction = node.style.direction || CSS_DIRECTION_LTR;
+    var skipLayout = 
+      !node.isDirty &&
+      node.lastLayout &&
+      node.lastLayout.requestedHeight === node.layout.height &&
+      node.lastLayout.requestedWidth === node.layout.width &&
+      node.lastLayout.parentMaxWidth === parentMaxWidth &&
+      node.lastLayout.direction === direction;
+      
+    if (skipLayout) {
+      node.layout.width = node.lastLayout.width;
+      node.layout.height = node.lastLayout.height;
+      node.layout.top = node.lastLayout.top;
+      node.layout.left = node.lastLayout.left;
+    } else {
+      if (!node.lastLayout) {
+        node.lastLayout = {};
+      }
+      
+      node.lastLayout.requestedWidth = node.layout.width;
+      node.lastLayout.requestedHeight = node.layout.height;
+      node.lastLayout.parentMaxWidth = parentMaxWidth;
+      node.lastLayout.direction = direction;
+      
+      // Reset child layouts
```

Comment:
Isn't it expected that all children of a node not have layout properties defined prior to processing that node? `layoutNodeImpl` seems to assume so. Without resetting, some [isUndefined checks](https://github.com/devongovett/css-layout/blob/js-caching/src/Layout.js#L718) might fail where they shouldn't, and the position of a node [is added](https://github.com/devongovett/css-layout/blob/js-caching/src/Layout.js#L457-L464) to it's position from the previous layout pass. I think all children have to be reset. The C version just does this after a layout pass instead of before the next one.